xlators/mgmt/glusterd/src/glusterd-rebalance.c

```diff
@@ -326,7 +327,11 @@ glusterd_handle_defrag_start(glusterd_volinfo_t *volinfo, char *op_errstr,
 
     sleep(5);
 
-    ret = glusterd_rebalance_rpc_create(volinfo);
+    if (!gf_is_service_running(pidfile, &pid)) {
```

Comment:
gf_is_service_running returns true if it is running but here is means we need to create rpc connection while it is not true means pid is not running, i think you need to change this.In case if process is not running we can't return to previous function we have to wait until process is not running, the end goal is we need to make a rpc connection with a rebalance daemon after start a rebalance daemon.

Reply:
Yeah. I think I mistook the return value part. ( I'll add the code part to back-off and try again if the process is not up ).

But the strange part is, while checking if this fix works, it actually worked. Not sure as to why rebalance status started updating properly in this change. :/